### PR TITLE
Secure download endpoint for secure messaging

### PR DESCRIPTION
**Pull request by AndriiHorishniiMOC**
Use the secure download endpoint for secure messaging instead of the default one.

[MOB-2023](https://glia.atlassian.net/browse/MOB-2023)

[MOB-2023]: https://glia.atlassian.net/browse/MOB-2023?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: widgetssdk/src/main/java/com/glia/widgets/filepreview/data/GliaFileRepositoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    private fun fetchFile(attachmentFile: AttachmentFile, callback: RequestCallback<InputStream>) {
+        val engagement = gliaCore.currentEngagement.orElse(null)
+        if (engagement == null && engagementConfigRepository.chatType == ChatType.SECURE_MESSAGING) {
+            secureConversations.fetchFile(attachmentFile.id, callback)
+        } else {
+            gliaCore.fetchFile(attachmentFile, callback)
+        }
+    }
+}
```
Comment (Pelkar) [inline widgetssdk/src/main/java/com/glia/widgets/filepreview/data/GliaFileRepositoryImpl.kt:104]: Note for future to help CR:
Make the change in the original java file, when feature has been reviewed and accepted, then translate to kotlin and retest before merger. This makes it better to review the specific changes only

**Review by Pelkar — APPROVED** [review]
Looks good, just added one comment for better review quality

**Review by andrews-moc — APPROVED** [review]

**Review by DavDo — APPROVED** [review]

**Comment by AndriiHorishniiMOC** [comment]
!squash

**Comment by AndriiHorishniiMOC** [comment]
!merge

**Pull request merged**